setup.py

```diff
@@ -1,32 +1,4 @@
-from distutils.core import setup
-import codecs
+from setuptools import setup
```

Comment:
The switch to `setuptools` seems quite significant to me. I am not sure I can reasonably test this in all possible scenarios to ensure it doesn't break anything.